config/webpack.config.js

```diff
@@ -11,152 +11,176 @@ var LodashModuleReplacementPlugin = require('lodash-webpack-plugin');
 var pluginUtils = require('../script/plugin_webpack_directories');
 var vendorEntry = require('./webpack.vendor');
 
-// must match config.webpack.dev_server.port
-var devServerPort = 3808;
 
-// set TARGETNODE_ENV=production on the environment to add asset fingerprints
-var production =
-  process.env.RAILS_ENV === 'production' ||
-  process.env.NODE_ENV === 'production';
 
-var bundleEntry = path.join(__dirname, '..', 'webpack/assets/javascripts/bundle.js');
+module.exports = env => {
+  // must match config.webpack.dev_server.port
+  var devServerPort = 3808;
 
-var plugins = pluginUtils.getPluginDirs();
+  // set TARGETNODE_ENV=production on the environment to add asset fingerprints
+  var production =
+    process.env.RAILS_ENV === 'production' ||
+    process.env.NODE_ENV === 'production';
 
-var resolveModules = [  path.join(__dirname, '..', 'webpack'),
-                        path.join(__dirname, '..', 'node_modules'),
-                        'node_modules/',
-                     ].concat(pluginUtils.pluginNodeModules(plugins));
+  var bundleEntry = path.join(__dirname, '..', 'webpack/assets/javascripts/bundle.js');
 
-var config = {
-  entry: Object.assign(
-    {
-      bundle: bundleEntry,
-      vendor: vendorEntry,
-    },
-    plugins.entries
-  ),
-  output: {
-    // Build assets directly in to public/webpack/, let webpack know
-    // that all webpacked assets start with webpack/
-
-    // must match config.webpack.output_dir
-    path: path.join(__dirname, '..', 'public', 'webpack'),
-    publicPath: '/webpack/',
-
-    filename: production ? '[name]-[chunkhash].js' : '[name].js'
-  },
-
-  resolve: {
-    modules: resolveModules,
-    alias: Object.assign({
-      foremanReact:
-        path.join(__dirname,
-           '../webpack/assets/javascripts/react_app'),
+  var plugins = pluginUtils.getPluginDirs();
+
+  var resolveModules = [  path.join(__dirname, '..', 'webpack'),
+                          path.join(__dirname, '..', 'node_modules'),
+                          'node_modules/',
+                       ].concat(pluginUtils.pluginNodeModules(plugins));
+
+  if (env && env.pluginName !== undefined) {
+    var pluginEntries = {};
+    pluginEntries[env.pluginName] = plugins['entries'][env.pluginName];
```

Comment:
Doesn't that mean it should call `plugin:assets:precompile[ansible]` or is the problem that the gem and node identifiers are different?

Reply:
Calling `plugin:assets:precompile[ansible]` doesn't work either, because it (correctly) mentions that it's `Unable to find registered plugin ansible`. I think the node identifiers in the hash above should be fixed to have the same names as the plugin gems, I'm working on it.